src/useDebouncedCallback.ts

```diff
@@ -286,6 +286,7 @@ export default function useDebouncedCallback<
     };
 
     func.cancel = () => {
+      const hadTimer = !!timerId.current;
       if (timerId.current) {
```

Comment:
These 2 lines re-use the same principle if `timerId.current` is defined.

Could you please combine them:
```
const  hadTimer = timerId.current;
if (hasTimer) {
...
```

and below you can keep:
```
if (hadTimer && forceUpdate) {
    forceUpdate({});
}
```
Motivation: there is really no point to force `hadTimer` to be strictly `boolean`, we want just to understand if it's empty or not, while `!!` adds excessive bytes (and use-debounce is optimised towards consuming as little space as possible).